src/ClaimRecipientManager.sol

```diff
@@ -0,0 +1,104 @@
+// SPDX-License-Identifier: BUSL-1.1
+
+pragma solidity 0.8.23;
+
+import { IClaimRecipientManager } from "./interfaces/IClaimRecipientManager.sol";
+import { IRegistrarLike } from "./interfaces/IRegistrarLike.sol";
+
+/**
+ * @title  Claim Recipient Manager contract for setting an returning claim recipients for Wrapped M Token yield.
+ * @author M^0 Labs
+ */
+contract ClaimRecipientManager is IClaimRecipientManager {
+    /* ============ Variables ============ */
+
+    /// @dev Registrar key prefix to determine the override recipient of an account's accrued yield.
+    bytes32 internal constant _CLAIM_OVERRIDE_RECIPIENT_PREFIX = "wm_claim_override_recipient";
+
+    /// @dev Registrar key of earners list.
+    bytes32 internal constant _CLAIM_RECIPIENT_ADMIN_LIST = "wm_claim_recipient_admins";
+
+    /// @inheritdoc IClaimRecipientManager
+    address public immutable registrar;
+
+    /// @dev Mapping of account to claim recipient.
+    mapping(address account => address recipient) internal _claimRecipients;
+
+    /* ============ Modifies ============ */
+
+    modifier onlyClaimRecipientAdmin() {
+        _revertIfNotClaimRecipientAdmin();
+        _;
+    }
+
+    /* ============ Constructor ============ */
+
+    /**
+     * @dev   Constructs the contract.
+     * @param registrar_ The address of a Registrar contract.
+     */
+    constructor(address registrar_) {
+        if ((registrar = registrar_) == address(0)) revert ZeroRegistrar();
+    }
+
+    /* ============ Interactive Functions ============ */
+
+    /// @inheritdoc IClaimRecipientManager
+    function setClaimRecipient(address account_, address recipient_) external onlyClaimRecipientAdmin {
+        _setClaimRecipient(account_, recipient_);
+    }
+
+    /// @inheritdoc IClaimRecipientManager
+    function setClaimRecipients(
+        address[] calldata accounts_,
+        address[] calldata recipients_
+    ) external onlyClaimRecipientAdmin {
+        if (accounts_.length != recipients_.length) revert ArrayLengthMismatch();
+
+        for (uint256 index_; index_ < accounts_.length; ++index_) {
+            _setClaimRecipient(accounts_[index_], recipients_[index_]);
+        }
+    }
+
+    /* ============ View/Pure Functions ============ */
+
+    /// @inheritdoc IClaimRecipientManager
+    function claimRecipientFor(address account_) external view returns (address recipient_) {
+        recipient_ = claimRecipientOverrideFor(account_);
+
+        // `claimRecipientOverrideFor` overrides `claimRecipientFor`.
+        return (recipient_ != address(0)) ? recipient_ : _claimRecipients[account_];
```

Comment:
Instead of handling it in `WrappedMToken.sol`, shouldn't this function return `account_` if `_claimRecipients[account_]` returns `address(0)`?

Reply:
It's really a separation of concerns. The claim recipient manager is just telling the wrapper what it knows. It's the wrapper's job to decide what a return of zero is. Even if we made this contract not return zero, we would still need, for safety reasons, a check to see what happens if the return is zero at the wrapper, so we might as well only do it there.